linked_list/implementation.py

```diff
@@ -1,40 +1,144 @@
 from .interface import AbstractLinkedList
-
+from .node import Node
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
-
+    
     def __init__(self, elements=None):
-        pass
-
+        self.start = None
+        self.end = None
+        self.elements = elements
+        if self.elements is not None:
+            for elem in self.elements:
+                self.append(elem)
+            
+    
     def __str__(self):
-        pass
+        if not self.start:
+            return "[]"
+        str_list='['
+        for node in self:
+            str_list+=str(node)
+            if node != self.end.elem:
+                str_list +=', '
+        str_list+=']'
+        return str_list
 
     def __len__(self):
-        pass
+        return self.count()
 
-    def __iter__(self):
-        pass
 
+    def __iter__(self):
+        if self.start is None:
+            raise StopIteration
+        node = None
+        while True:
+            if node is None:
+                node = self.start
+            yield node.elem
+            node = node.next
+            if node is None:
+                raise StopIteration
+        
     def __getitem__(self, index):
-        pass
+        counter = 0
+        for node in self:
+            if counter == index:
+                return node
+            counter += 1
 
     def __add__(self, other):
-        pass
+        new_list = LinkedList()
+        for elem in self:
+            new_list.append(elem)
+        for node in other:
+            new_list.append(node)
+        return new_list
+        
 
     def __iadd__(self, other):
-        pass
+        if not other.start:
+            return self
+        for node in other:
+            self.append(node)
+        return self
+
+
+    def __ne__(self, other):
+        if self.count() != other.count():
+            return True
+        for x, y in zip(self, other):
+            if x!=y:
+                return True
+        return False
 
     def __eq__(self, other):
-        pass
+        if not self.start and not other.start:
+            return True
+        if self.count() != other.count():
+            return False
+        for x, y in zip(self, other):
+            if x!=y:
+                return False
+        return True
+
 
     def append(self, elem):
-        pass
+        node = Node(elem)
+        if self.start is None:
+            self.start = node
+        if self.end is None:
+            self.end = node
+            return
+    
+        self.end.next = node
+        self.end = node
 
+        
     def count(self):
-        pass
+        i = 0
+        for x in self:
+            if x is None:
```

Comment:
Just curious, since iter returns elem and raises StopIteration, would x be None in any case?